vector/src/main/java/im/vector/app/features/userdirectory/UserDirectoryUserItem.kt

```diff
@@ -40,15 +43,23 @@ abstract class UserDirectoryUserItem : VectorEpoxyModel<UserDirectoryUserItem.Ho
     override fun bind(holder: Holder) {
         super.bind(holder)
         holder.view.setOnClickListener(clickListener)
-        // If name is empty, use userId as name and force it being centered
-        if (matrixItem.displayName.isNullOrEmpty()) {
-            holder.userIdView.visibility = View.GONE
-            holder.nameView.text = matrixItem.id
-        } else {
-            holder.userIdView.visibility = View.VISIBLE
-            holder.nameView.text = matrixItem.displayName
-            holder.userIdView.text = matrixItem.id
+        val displayName = matrixItem.displayName
+        when {
+            TchapUtils.isExternalTchapUser(matrixItem.id) -> {
+                holder.nameView.text = matrixItem.id
```

Comment:
here the displayname is used (not the matrix id), `holder.nameView.text = displayName`

Reply:
Better!
<img width="334" alt="image" src="https://user-images.githubusercontent.com/11990514/124253498-5f5c6e80-db28-11eb-85c4-b5f1ed7cde10.png">